split-webapp/split/src/context/UserContext.js

```diff
@@ -0,0 +1,39 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { getCookie } from "../utils/helpers";
+
+export const UserContext = React.createContext({});
+
+export class UserProvider extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      username: getCookie("username")
```

Comment:
We would now have duplicate state rather than a single source of truth.

I was wondering: Would it would be better to query the cookies directly whenever we need it, rather than store it in a state and open the opportunity for it to get out of sync with the cookie?

But for our simple app, this is probably fine. And like you said, this context API makes it easy to extend the user context in the future.

Reply:
Good point about duplicate state. I chose to do it this way as I thought it would be good to make the actual components unaware of where the user state is actually coming from. This also allows us to change the source of this state more easily (i.e cookie, returned in response from login request) if multiple components are depending on it. 

I'm happy to change it if you think the trade-offs aren't worth it :D If anyone else has some thoughts feel free to chime in.

Reply:
> I'm happy to change it if you think the trade-offs aren't worth it

Nah - what you have right now is probably worth it c: